compatible/src/main/java/io/seata/integration/tx/api/interceptor/handler/GlobalTransactionalInterceptorHandler.java

```diff
@@ -16,80 +16,53 @@
  */
 package io.seata.integration.tx.api.interceptor.handler;
 
-import io.seata.spring.annotation.GlobalLock;
-import io.seata.spring.annotation.GlobalTransactional;
-import io.seata.tm.api.FailureHandler;
 import org.apache.seata.core.model.GlobalLockConfig;
 import org.apache.seata.integration.tx.api.annotation.AspectTransactional;
-import org.apache.seata.integration.tx.api.interceptor.InvocationWrapper;
-import org.apache.seata.integration.tx.api.util.ClassUtils;
-import org.apache.seata.rm.GlobalLockExecutor;
+import org.apache.seata.spring.annotation.GlobalLock;
+import org.apache.seata.spring.annotation.GlobalTransactional;
+import org.apache.seata.tm.api.FailureHandler;
 
 import java.lang.reflect.Method;
 import java.util.Set;
 
 /**
  * The type Global transactional interceptor handler.
  */
+@Deprecated
 public class GlobalTransactionalInterceptorHandler extends org.apache.seata.integration.tx.api.interceptor.handler.GlobalTransactionalInterceptorHandler {
 
-
     public GlobalTransactionalInterceptorHandler(FailureHandler failureHandler, Set<String> methodsToProxy) {
         super(failureHandler, methodsToProxy);
     }
 
-    public GlobalTransactionalInterceptorHandler(FailureHandler failureHandler, Set<String> methodsToProxy, AspectTransactional aspectTransactional) {
+    public GlobalTransactionalInterceptorHandler(FailureHandler failureHandler, Set<String> methodsToProxy,
```

Comment:
FailureHandler is a user API; either perform type conversion beforehand, or use the io.seata package.

Reply:
> FailureHandler is a user API; either perform type conversion beforehand, or use the io.seata package.

我通过扩展了apacheseata failureHandler，并将原始failureHandler至为泛型，所以入参为org apache的failureHandler不会对ioseata的failureHandler扩展造成影响，因为ioseata的failureHandler继承至orgapache的failureHandler，而不是apacheseatafailureHandler这个类
I extended apacheseata failureHandler and generalized the original failureHandler so that failureHandler from org apache doesn't affect the extension of ioseata's failureHandler because ioseata's failureHandler inherits from orgapache's failureHandler, not from apacheseatafailureHandler.